docs/src/main/asciidoc/microprofile-graphql.adoc

```diff
@@ -0,0 +1,416 @@
+
+Add the following to our `FilmResource` class.
+
+----
+    public List<Hero> heroes(@Source Film film) { // <1>
+    return service.getAllHeroes();
+    }
+----
+
+<1> Enable `List<Hero>` data to be added to queries that response with `Film`
+
+By adding this method we have effectively changing the schema of the GraphQL API.
+Although the schema has changed the previous queries will still work.
+Since we only expanded the API to be able to retrieve `Hero` data with `Film` data.
+
+Enter the following into `graphiql` and to retrieve the film and hero data.
+
+----
+query getFilmHeroes {
+  film(filmId: 1) {
+    title
+    director
+    releaseDate
+    episodeID
+    heroes {
+      name
+      height
+      mass
+      darkSide
+      lightSaber
+    }
+  }
+}
+----
+
+The response now includes the heroes of the film.
+
+== Mutations
+
+Mutations are used when data is create, updated or deleted.
+
+Lets now add the ability to add and delete heroes to our GraphQL API.
+
+Add the following to our `FilmResource` class:
+
+----
+    @Mutation
+    public Hero updateHero(Hero hero) {
+        service.createHero(hero);
+        return hero;
+    }
+
+    @Mutation
+    public Hero deleteHero(int id) {
+        return service.deleteHero(id);
+    }
+----
+
+Enter the following into `graphiql` and to insert the last film of the original trilogy.
+
+----
+mutation addHero {
+  updateHero(hero: {
+      name: "Han",
+      surname: "Solo"
+      height: 1.85
+      mass: 80
+      isDarkSide: false
+      episodeIds: [4, 5, 6]
+  	}
+  )
+  {
+    name
+    surname
+  }
+}
+----
+
+By using this mutation we have created a hero entity in our service.
+
+Notice how in the response we have retrieved the `name` and `surname`
+of the created Hero. This is because we selected to retrieve
+these fields in the response within the `{ }` in the mutation query.
+This can easily be a server side generated field that the client may require.
+
+Lets now try deleting an entry
+
+----
+mutation DeleteHero {
+  deleteHero(id :3){
+    name
+    surname
+  }
+}
+----
+
+Similar to the `updateHero` mutation method we also retrieve the `name` and
+`surname` of the hero we have deleted which is defined in `{ }`.
+
+Queries can also be done on individual fields. For example, lets
+create a method to query heroes by there last name.
+
+----
+    @Query
+    public List<Hero> getHeroesWithSurname(@DefaultValue("Skywalker") String surname) {
+        return service.getHeroesBySurname(surname);
+    }
+----
+
+By using the `@DefaultValue` annotation we have determined the surname value when the parameter
+not given.
+
+Test the following queries with `graphiql`
+
+----
+query heroWithDefaultSurname {
+  heroesWithSurname{
+    name
+    surname
+    lightSaber
+  }
+}
+query heroWithSurnames {
+  heroesWithSurname(surname: "Vader") {
+    name
+    surname
+    lightSaber
+  }
+}
+----
+
+== Conclusion
```

Comment:
I think this section is missing some meat :-)

Reply:
Will complete as soon as possible. :)